### DISCUSSION TITLE
With a dum-based app how to add custom headers in response?

### DISCUSSION THREAD

**Discussion by alienpenguin** [resiprocate/resiprocate#322] [Q&A] [answered]
Hello all,
i have a dum-based app and i am not figuring how to add custom SIP headers in response messages...

with the ServerInviteSessionHandle i am able to set the provdedAnswer and answer with accept() method or send a provisional with provisional(180);
but how can i access the messages that are going to be sent to add a custom header?

thanks for the insight

**Comment by portsip** [comment, marked as answer]
OnReadyToSend

On Sat, Aug 5, 2023 at 01:14 alienpenguin ***@***.***> wrote:

> Hello all,
> i have a dum-based app and i am not figuring how to add custom SIP headers
> in response messages...
>
> with the ServerInviteSessionHandle i am able to set the provdedAnswer and
> answer with accept() method or send a provisional with provisional(180);
> but how can i access the messages that are going to be sent to add a
> custom header?
>
> thanks for the insight
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/resiprocate/resiprocate/discussions/322>, or
> unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ACXARLTFGPZP3D4S6WPP7XTXTUUYBANCNFSM6AAAAAA3EPGUFE>
> .
> You are receiving this because you are subscribed to this thread.Message
> ID: ***@***.***>
>
-- 
Marc Janko
Sales Manager, PortSIP Solutions, Inc.

*Website *www.portsip.com  |  *Email ***@***.***

*Click here to submit a request
<http://portsip.zendesk.com/hc/en-us/requests/new>     **Knowledge Bas
<https://support.portsip.com/portsip-pbx-user-guide/portsip-pbx-administration-guide-v16.x>e*

*PortSIP PBX is not based on Asterisk or Freeswitch*

**Reply by alienpenguin** [reply]
thanks, i was overlooking that method :)

**Comment by sgodin** [comment]
More info here:  https://github.com/resiprocate/resiprocate/wiki/Modifying-Decorating-messages-sent
